### ISSUE TITLE
Being able to install "arbitrary" gitlab ci ghc artifacts 

### ISSUE THREAD

**Issue by jneira** [Mistuke/GhcChoco#11]
Hi, i was thinking how we could test [lastest ghc-8.10 branch windows ghc against haskell-language-server](https://gitlab.haskell.org/ghc/ghc/-/issues/18496#note_314035) and i've wondered if chocolatey could install the generated artifacts in gitlab ci.

The process should take some sort of key to identify the artifact:
* a commit to get the lastest artifact asociated with
* a branch name like `ghc-8.10` to get its last artifact
* a ghc version included the date (as i've observed artifacts follow the convention ghc-8.10.1-20101120)

Then we could identify the artifact using the gitlab api (is it feasible? dont know), download it and follow the steps for normal releases

It could serve to make easy test early ghc versions against downstream packages like hls, where we could keep an optional job using the latest artifact of each supported major version.

Maybe is it already possible? maybe is only a crazy idea?

**Comment by Mistuke** [comment]
> Maybe is it already possible? maybe is only a crazy idea? 
Yes it's actually not that crazy of an idea :) I have already had a talk about this with @bgamari in the past.

He have me a url for the nightlies but I think I have lost it....

I already have the infrastructure in place to produce these (I used to produce them for master)
https://www.myget.org/feed/mistuke/package/nuget/ghc-head

These can be added as a source to choco. (since they need to be removeable I don't want to use the main repo for this)
I have stopped publishing these as their download counts were always 0 (though I do still build and store GHC nightlies in my Azure account, I seem to have 3+ years of nightlies...)

Using gitlab's binaries is a good idea.. if we can figue out the urls it would be easy to do.

**Comment by jneira** [comment]
Many thanks for take care. :slightly_smiling_face: 

> I have stopped publishing these as their download counts were always 0 

Well, the projects i am collaborating lately are very tied to ghc internals (hls, ghcide, etc) and, in general, quite packages/projects need several months to adapt after a ghc major or minor release, if the latter supposes a `base` or other boot libraries breaking bump.
In this case to have available head for ci could help to give the opportunity to them for advance needed changes. But well, time is a scarce resource and it seems that is not what happens normally.
  
But test against the head "fix" version of any living minor version (currently 8.10.x) release would be great. I would add it to ci in hls/ghcide for example.

**Comment by Mistuke** [comment]
Makes sense. @bgamari is there nightlies or release branches too? If so where can I find the artifacts?
